pdf.go

```diff
@@ -3,12 +3,110 @@ package docconv
 import (
 	"fmt"
 	"io"
+	"io/ioutil"
 	"log"
+	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
+	"sync"
 	"time"
 )
 
+var (
+	exts = []string{".jpg", ".tif", ".tiff", ".png"}
+)
+
+var pdfMutex struct {
+	sync.Mutex
+}
+
+func compareExt(ext string, exts []string) bool {
+	for _, e := range exts {
+		if ext == e {
+			return true
+		}
+	}
+	return false
+}
+
+func PDFImages(path string) (string, map[string]string, error) {
+	tmp, err := ioutil.TempDir("/tmp", "tmp-imgs-")
+	if err != nil {
+		log.Println(err)
+		return "", nil, err
+	}
+	tmpDir := fmt.Sprintf("%s/", tmp)
+	defer os.RemoveAll(tmpDir)
+
+	_, err = exec.Command("pdfimages", "-j", path, tmpDir).Output()
+	if err != nil {
+		return "", nil, err
+	}
+
+	files := []string{}
+
+	walkFunc := func(path string, info os.FileInfo, err error) error {
+		path, err = filepath.Abs(path)
+		if err != nil {
+			return err
+		}
+
+		if compareExt(filepath.Ext(path), exts) == true {
+			files = append(files, path)
+		}
+		return nil
+	}
+	filepath.Walk(tmpDir, walkFunc)
+
+	var wg sync.WaitGroup
+	m := make(map[int]string)
```

Comment:
Better to create an anonymous type here with the map and mutex inside, rather than having a global `pdfMutex` which is only used in one place.